packages/text-field/Input.tsx

```diff
@@ -130,19 +132,19 @@ export default class Input<T extends HTMLElement = HTMLInputElement> extends Rea
       setInputId && setInputId(id!);
     }
 
-    if (value !== prevProps.value) {
+    if ((value !== prevProps.value || foundationChanged) && foundation) {
       handleValueChange && handleValueChange(value, () => {
-        // only call #foundation.setValue on programatic changes;
-        // not changes by the user.
-        if (this.state.wasUserTriggeredChange) {
+        // #foundation.setValue should only be called on programatic changes,
+        // not changes by the user, nor when the foundation changes.
+        if (this.state.wasUserTriggeredChange && !foundationChanged) {
```

Comment:
This reads better, but can you add a unit test for this case?

Reply:
sure!